src/WebJobs.Extensions.DocumentDB/Trigger/CosmosDBTriggerAttributeBindingProvider.cs

```diff
@@ -91,27 +92,55 @@ public Task<ITriggerBinding> TryCreateAsync(TriggerBindingProviderContext contex
 
         private string ResolveAttributeConnectionString(CosmosDBTriggerAttribute attribute)
         {
-            if (string.IsNullOrEmpty(_monitorConnectionString) &&
-                string.IsNullOrEmpty(attribute.ConnectionStringSetting))
+            string connectionString = this._monitorConnectionString;
+            if (!TryReadFromSettings(attribute.ConnectionStringSetting, connectionString, out connectionString))
+            {
+                throw new InvalidOperationException(
+                    string.Format(CultureInfo.CurrentCulture,
+                    "The CosmosDBTriggerAttribute.ConnectionStringSetting '{0}' property specified does not exist in the Application Settings.",
+                    attribute.ConnectionStringSetting));
+            }
+
+            if (string.IsNullOrEmpty(connectionString))
             {
                 throw new InvalidOperationException(
                     string.Format(CultureInfo.CurrentCulture,
-                    "The DocumentDBTrigger connection string must be set either via a '{0}' app setting, via the DocumentDBTriggerAttribute.ConnectionString property or via DocumentDBConfiguration.TriggerConnectionString.",
+                    "The CosmosDBTrigger connection string must be set either via a '{0}' app setting, via the CosmosDBTriggerAttribute.ConnectionStringSetting property or via DocumentDBConfiguration.ConnectionString.",
                     DocumentDBConfiguration.AzureWebJobsDocumentDBConnectionStringName));
             }
 
-            return attribute.ConnectionStringSetting ?? _monitorConnectionString;
+            return connectionString;
         }
 
         private string ResolveAttributeLeasesConnectionString(CosmosDBTriggerAttribute attribute, string triggerConnectionString)
         {
             // If the leases connection is not specified, it connects to the monitored service
-            return attribute.LeaseConnectionStringSetting ?? _leasesConnectionString ?? triggerConnectionString;
+            string connectionString = string.IsNullOrEmpty(this._leasesConnectionString) ? triggerConnectionString : this._leasesConnectionString;
+            if (!TryReadFromSettings(attribute.LeaseConnectionStringSetting, connectionString, out connectionString))
+            {
+                throw new InvalidOperationException(
+                    string.Format(CultureInfo.CurrentCulture,
+                    "The CosmosDBTriggerAttribute.LeaseConnectionStringSetting '{0}' property specified does not exist in the Application Settings.",
+                    attribute.ConnectionStringSetting));
+            }
```

Comment:
Same as the other comment, it's meant to catch the specify scenario of an Attribute with an non-existant Setting in the environment.
The more complete message sounds aimed at a scenario where no Connection string was specified anywhere.